protocol.go

```diff
@@ -287,3 +287,12 @@ func (p *Protocol) PullTransactions(
 
 	return res, nil
 }
+
+func (p *Protocol) GetBlock(ctx context.Context, req *GetBlockRequest) (*GetBlockResponse, error) {
```

Comment:
Rather than add a separate `GetBlockRequest/GetBlockResponse`, care to use just `QueryBlockRequest/QueryBlockResponse`?

Reply:
there is no `QueryBlockRequest` but there is `QueryRequest` and it contains field `cache_block_id` along with block index. same for response object, one used for querying has additional field in it.
i think its better to use different message objects here.

Reply:
Hm, I'd say it's best to just stick to 1 message object to make the RPC interface as minimal as possible. We can intentionally leave out cache_block_id here in this case.

Reply:
i don't think that reusing messages that are specific for some particular request is a good idea.
query messages already have extra information and specific name, if we'll need to add something more to them, then using those in `GetBlock` will be even more confusing.

Reply:
QueryBlock is equivalent to asking for a block from a peer though - the only additional field is for caching.